.github/workflows/rust-test.yml

```diff
@@ -36,6 +36,24 @@ jobs:
         if: ${{ matrix.os == 'macos-latest' }}
         run: brew install llvm libomp cmake
 
+      - name: Setup noir cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/noir_cache/
+          key: noir_cache
```

Comment:
@kevaundray I was wondering if there is anything wrong with us caching the noir_cache in GitHub Actions cache? It seems like it restores very quickly and would probably cut down on our spurious mac failures.

Reply:
Yeah we could cache it -- For the future, we could also say that we cache all setup files for backends that have first class support

Reply:
Sounds good. I've updated to clean up some logic, so we're good for a review 👍